File: atm cajero sockets/cajero-servidor/src/atm/comun/protocolo.py
"""
Contrato de mensajes que viajan por el socket entre el cajero (cliente)
y el banco (servidor). Este archivo existe también, palabra por palabra,
en cajero-cliente/src/atm/comun/protocolo.py: son dos proyectos
independientes (como clientsocket/ y socketserver/ del profesor), así
que no comparten import, solo se ponen de acuerdo en este "contrato".
"""

OP_DEPOSITAR = "DEPOSITAR"
OP_RETIRAR = "RETIRAR"
OP_CONSULTAR = "CONSULTAR"
OP_SALIR = "SALIR"


def crear_peticion(operacion: str, numero_cuenta: str, clave: str, monto: float = None) -> dict:
    return {
        "operacion": operacion,
        "numero_cuenta": numero_cuenta,
        "clave": clave,
        "monto": monto,
    }


def crear_respuesta(exitoso: bool, mensaje: str, numero_cuenta: str, saldo) -> dict:
    return {
        "exitoso": exitoso,
        "mensaje": mensaje,
        "numero_cuenta": numero_cuenta,
        "saldo": saldo,
    }

File: atm cajero sockets/cajero-cliente-go/src/atm/comun/protocolo.go
// Contrato de mensajes que viajan por el socket entre el cajero (cliente)
// y el banco (servidor). Existe también, campo por campo, en el
// protocolo.py de cajero-servidor y cajero-cliente: son proyectos
// independientes que solo se ponen de acuerdo en este "contrato" (mismas
// claves JSON: operacion, numero_cuenta, clave, monto / exitoso, mensaje,
// numero_cuenta, saldo).
package comun

const (
	OpDepositar = "DEPOSITAR"
	OpRetirar   = "RETIRAR"
	OpConsultar = "CONSULTAR"
	OpSalir     = "SALIR"
)

type Peticion struct {
	Operacion    string   `json:"operacion"`
	NumeroCuenta string   `json:"numero_cuenta"`
	Clave        string   `json:"clave"`
	Monto        *float64 `json:"monto"`
}

type Respuesta struct {
	Exitoso      bool     `json:"exitoso"`
	Mensaje      string   `json:"mensaje"`
	NumeroCuenta string   `json:"numero_cuenta"`
	Saldo        *float64 `json:"saldo"`
}

func CrearPeticion(operacion, numeroCuenta, clave string, monto *float64) Peticion {
	return Peticion{
		Operacion:    operacion,
		NumeroCuenta: numeroCuenta,
		Clave:        clave,
		Monto:        monto,
	}
}
